cmd/humanlog/genny.go

```diff
@@ -90,18 +99,79 @@ func genny(
 
 	now := start
 	for {
-		if err := emitMessage(out, now, src); err != nil {
-			return err
-		}
-		nextArrival := arrivalRateDist.Rand()
-		nextMsgIn := time.Duration(float64(interval) / nextArrival)
 		select {
 		case <-ctx.Done():
 			return nil
-		case <-time.After(nextMsgIn):
+		default:
+			nextArrival := arrivalRateDist.Rand()
+			nextMsgIn := time.Duration(float64(interval) / nextArrival)
+			time.Sleep(nextMsgIn)
 			now = now.Add(nextMsgIn)
+
+			if err := emitLog(out, now, src, format); err != nil {
+				return err
+			}
+		}
+	}
+}
+
+func emitLog(out io.Writer, now time.Time, src rand.Source, format string) error {
+	var log string
+	switch format {
+	case "logfmt":
+		log = generateLogfmtLog(now, src)
+	case "json":
+		log = generateJSONLog(now, src)
+	case "otel":
+		log = generateOtelLog(now, src)
+	case "mixed":
+		switch randel(src, []string{"logfmt", "json", "otel"}) {
```

Comment:
You could use recursion here to avoid having to maintain two levels of `switch` with the same keys`